compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -1188,9 +1189,19 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         unsafe { llvm::LLVMBuildZExt(self.llbuilder, val, dest_ty, UNNAMED) }
     }
 
-    fn do_not_inline(&mut self, llret: &'ll Value) {
-        let noinline = llvm::AttributeKind::NoInline.create_attr(self.llcx);
-        attributes::apply_to_callsite(llret, llvm::AttributePlace::Function, &[noinline]);
+    fn apply_attrs_to_cleanup_callsite(&mut self, llret: &'ll Value) {
+        let mut attrs = SmallVec::<[_; 2]>::new();
+
+        // Cleanup is always the cold path.
+        attrs.push(llvm::AttributeKind::Cold.create_attr(self.llcx));
+
+        // In LLVM versions with deferred inlining (currently, system LLVM < 14),
+        // inlining drop glue can lead to exponential size blowup, see #41696 and #92110.
+        if !llvm_util::is_rust_llvm() && llvm_util::get_version() < (14, 0, 0) {
```

Comment:
The is_rust_llvm check shouldn't be necessary.

Reply:
Right.

Also, I regret to inform you--it seems like there are more inlining issues.
Compiling https://github.com/MarkDDR/long_compile_rustc_1.59.0 with this patch takes nearly as long as it does with 1.59.0.